### PR TITLE
[UWP] Implement CurrentItem on CarouselView

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement **CurrentItem** on CarouselView (**UWP**).

carousel-uwp-horizontal-currentitem
carousel-uwp-vertical-currentitem

### API Changes ###

 None

### Platforms Affected ### 
- UWP

### Behavioral/Visual Changes ###
With this changes, the CurrentItem must be updated when scrolling the Carousel.

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
Launch Core Gallery, navigate to Carousel Core samples and scroll the Carousel. The slider with the position must be updated.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jfversluis** [comment]
I'm guessing this needs to go into 4.3 rather than master? :)

File: Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewStyles.xaml [hunk 1]
```diff
@@ -7,6 +7,10 @@
         <VirtualizingStackPanel Orientation="Horizontal" />
     </ItemsPanelTemplate>
 
+    <ItemsPanelTemplate x:Key="VerticalListItemsPanel">
+        <VirtualizingStackPanel Orientation="Vertical" />
```
Comment (hartez) [inline Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewStyles.xaml]: This should be an `ItemsStackPanel` rather than a `VirtualizingStackPanel`.

**Comment by rmarinho** [comment]
Can you rebase on top of 4.3.0 changes @jsuarezruiz ? thanks

**Comment by jfversluis** [comment]
@jsuarezruiz seems like a rebase is needed here!

File: Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewStyles.xaml (hunk 1 above)
Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewStyles.xaml]: Changed!.

**Comment by jsuarezruiz** [comment]
I have updated the branch (rebase and simplified the logic to update CurrentItem on UWP).

File: Xamarin.Forms.Core/Items/CarouselView.cs [hunk 2]
```diff
@@ -165,6 +165,9 @@ public LinearItemsLayout ItemsLayout
 			set => SetValue(ItemsLayoutProperty, value);
 		}
 
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public bool IsScrolling { get; set; }
```
Comment (hartez) [inline Xamarin.Forms.Core/Items/CarouselView.cs]: Does `IsScrolling` ever have a different value than `IsDragging`? I'm wondering if we need two separate values, or if we could just have one.

Comment (jsuarezruiz) [reply Xamarin.Forms.Core/Items/CarouselView.cs]: Yes, the main difference is:

- IsScrolling: Indicates when the CarouselView is scrolling independently if it is started from ScrollTo  method (programmatically) or from a user interaction.
- IsDragging: Indicates when the user is dragging the Carousel. The Carousel may be scrolling while IsDragging is false.

**Pull request merged**